t/t7407-submodule-foreach.sh

```diff
@@ -349,11 +349,26 @@ test_expect_success 'use "git clone --recursive" to checkout all submodules' '
 	)
 '
 
+test_expect_success 'use "git clone" with submodule.recurse=true to checkout all submodules' '
+	git clone -c submodule.recurse=true super clone5 &&
+	(
+		cd clone5 &&
+		git rev-parse --resolve-git-dir .git &&
+		git rev-parse --resolve-git-dir sub1/.git &&
+		git rev-parse --resolve-git-dir sub2/.git &&
+		git rev-parse --resolve-git-dir sub3/.git &&
+		git rev-parse --resolve-git-dir nested1/.git &&
+		git rev-parse --resolve-git-dir nested1/nested2/.git &&
+		git rev-parse --resolve-git-dir nested1/nested2/nested3/.git &&
+		git rev-parse --resolve-git-dir nested1/nested2/nested3/submodule/.git
+	)
```

Comment:
It would be better to do this in a single invocation, and then to compare it to the expected values. Besides, I don't think that it needs to be exhaustive. So I would do this instead:

```sh
git -C clone5 rev-parse \
    --resolve-git-dir .git \
    --resolve-git-dir nested1/nested2/nested3/submodule/.git >actual &&
cat >expect <<-EOF &&
.git
$(pwd)/nested1/nested2/nested3/submodule/.git
EOF
test_cmp expect actual
```

Reply:
Sorry, don’t get it. I just copied the test for ‘git clone –recursive’. This command should do the identical job like git clone -c submodule.recurse=true’. Why do different tests?

Reply:
Because you can improve the style? You also do not need to be _so_ extensive in an essentially slightly repeated test. Testing whether _one_ nested submodule was cloned should be _plenty_ enough.

Reply:
OK, I will take a closer look. I like to understand the test regression code a bit better first